### DISCUSSION TITLE
Optimize the ABP to fully support fast path?

### DISCUSSION THREAD

**Discussion by watsonsong** [Sixze/ALS-Refactored#474] [Q&A]
Is it will be better performance if turn on the "Warn About Ablueprint Usage" and fix all warning. Cache the conditions check and make the ABP fully support fast path?

**Comment by Sixze** [comment]
It will be better, but at the same time the animation blueprint will become more rigid. I believe that such optimizations should be done in the real project and not in the plugin, but I will think about it.

**Reply by watsonsong** [reply]
What if we calculate the condition in event graph by 'Blueprint Thread Safe Update Animation'. Check the tag and logics and transform them into bool variable. Move the condition from Animation Graphs into 'Blueprint Thread Safe Update Animation' instead of in C++. And make Animation Graphs keep fast path.
Will it help performance? I am not sure.

**Reply by Sixze** [reply]
No, it won't and it will likely make things worse. The purpose of Fast Path is to avoid Blueprint VM overhead, and moving conditions to Blueprint Thread Safe Update Animation will not change that.
